.gitignore

```diff
@@ -79,3 +79,6 @@ node_modules/
 
 # Generated JavaScript Bundles
 jsbundles
+
+# jenkins generated npm lock files at build time, however that should not be checked in ever
+package-lock.json
```

Comment:
Jenkins shouldn't generate a package-lock file as it uses `yarn`

Reply:
I am not sure how it is generated, but yeah since we use yarn the `package-lock.json` should never be checked in anyhow, so I think it not a bad idea to include it.

Reply:
@basil not sure what you mean, the `package-lock.json` is not in our repo however sometimes I find it listed under `Untracked files:`, it will be normally generate by running `npm install` however in the context of having https://github.com/jenkinsci/jenkins/blob/master/yarn.lock this file should never be added to git ever.

Reply:
https://github.com/jenkinsci/jenkins/pull/9757#discussion_r1768144800

The comment needs updating then if you want to keep this, its not Jenkins generating it